src/views/discover/discover.hooks.ts

```diff
@@ -161,12 +170,22 @@ export function useDiscoverCollectionsByVolumeLazyQueryWithTransforms(
   fetchMoreLimit: number
 ): DiscoverCollectionsQueryContext {
   const mergeResultsFunction: UpdateResultsFunction<DiscoverCollectionsByVolumeQuery> = useCallback(
-    (previous, more) => {
+    (previous, more, setHasMore) => {
       if (!more) return previous;
+
+      if (setHasMore) {
+        if (more.collectionsFeaturedByVolume.length === 0) {
+          setHasMore(false);
+        } else {
+          setHasMore(true);
+        }
+      }
```

Comment:
This should already be taken care of by `useHolaplexInfiniteScrollQuery`. Sorry, should have added some docs to that, but it's main purpose is to take care of lazy loading more rows. It returns a stateful variable `hasMore` that gets updated when the lazy loading is triggered.

Reply:
it wasn't handling it correctly.

Reply:
Ah, great catch! I would love if, rather than try to put in a bandaid here, we could make `useHolaplexInfiniteScrollQuery` work correctly. I'm happy to take a stab at it